core/src/main/java/com/linecorp/armeria/client/DefaultHttpClient.java

```diff
@@ -74,8 +74,9 @@ private HttpResponse execute(@Nullable EventLoop eventLoop, HttpRequest req) {
         }
 
         if (isUndefinedUri(uri())) {
-            req.abort();
-            return HttpResponse.ofFailure(new IllegalArgumentException("no authority: " + req.path()));
+            final IllegalArgumentException cause = new IllegalArgumentException("no authority: " + req.path());
+            req.abort(cause);
```

Comment:
I think this might best help me understand the different use cases of `abort` and `close`, which I'm still not comfortable about (I'm still not sure we need two methods).

This PR was about allowing customizing of response timeout, but it looks like we're changing existing behavior not related to timeout. So I wonder, what is the result of this change for the user? Did something change / get better by passing `cause` to `abort` everywhere? Usability-wise, I think it's worse to have this sort of duplicated logic, so I wonder what the effect is.

Reply:
> This PR was about allowing customizing of response timeout, but it looks like we're changing existing behavior not related to timeout.

I agree with your opinion. This PR is not just about adding a response timeout handler. To customize the response timeout handler, we need a new abort method that consumes an abort cause. While adding the method, we wanted to migrate and fix the existing abort logic. Sorry for the confusion. I think the adding `abort(cause)` issue should be handled first. 

> Usability-wise, I think it's worse to have this sort of duplicated logic, so I wonder what the effect is.

This seems to have duplicated logic, but I believe we could deliver more detail abort cause to `log.responseCause()`, `log.requestCause()`, and `completionFuture()`.

Reply:
It actually fixes a bug where the specified request is never consumed when a bad request is given. This may yield a memory leak, so he fixed it by aborting the request. Besides this, I don't see any behavioral changes, right?

Reply:
We were already calling `req.abort()` without a cause so I wouldn't expect any difference in terms of memory leak. From what I understand, now `log.requestCause` will have the actual cause, which is the same as `log.responseCause` when before it was just `AbortedStreamException` for the request. While it's sort of double-logging, I guess it's still more useful than generic `AbortedStreamException` so it makes sense. I sort of wish returning a failed response automatically closed the request but that's not so related to this PR.

Reply:
Ahh. Sorry, I was seeing a wrong diff. :sweat_smile: